GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -164,17 +168,29 @@ public static ArgumentString RevertCmd(ObjectId commitId, bool autoCommit, int p
 
         public static ArgumentString ResetSoftCmd(string commit)
         {
-            return "reset --soft \"" + commit + "\"";
+            return new GitArgumentBuilder("reset")
+            {
+                "--soft",
+                commit?.QuoteNE()
```

Comment:
null check is unnecessary, `QuoteNE` handles `null`s
the same applies to other instances

Reply:
Ok.  I saw it in other places.